src/main/java/appeng/helpers/DualityInterface.java

```diff
@@ -553,17 +548,20 @@ public DimensionalCoord getLocation()
 		return new DimensionalCoord( this.iHost.getTileEntity() );
 	}
 
-	public IInventory getInternalInventory()
+	public IItemHandlerModifiable getInternalInventory()
 	{
 		return this.storage;
 	}
 
 	public void markDirty()
 	{
-		for( int slot = 0; slot < this.storage.getSizeInventory(); slot++ )
-		{
-			this.onChangeInventory( this.storage, slot, InvOperation.markDirty, ItemStack.EMPTY, ItemStack.EMPTY );
-		}
+		// TODO: should not be needed
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1135)